### PR TITLE
WIP Upgrade to minimesos 0.5.0

### PR DESCRIPTION
**Pull request by frankscholten**

### CODE REVIEW

File: system-test/src/main/java/org/apache/mesos/elasticsearch/systemtest/Configuration.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
     private String schedulerName = "elasticsearch-scheduler";
     private int schedulerGuiPort = 31100;
     private int elasticsearchNodesCount = 3;
-    private int elasticsearchMemorySize = 512;
+    private int elasticsearchMemorySize = 250;
```
Comment (sadovnikov) [inline system-test/src/main/java/org/apache/mesos/elasticsearch/systemtest/Configuration.java]: Should it be 256 ?

File: system-test/src/main/java/org/apache/mesos/elasticsearch/systemtest/Main.java [hunk 2]
```diff
@@ -56,12 +59,12 @@ private static void seedData(MesosCluster cluster, ElasticsearchSchedulerContain
         String taskHttpAddress;
         try {
             ESTasks esTasks = new ESTasks(TEST_CONFIG, schedulerContainer.getIpAddress());
-            new TasksResponse(esTasks, cluster.getConfig().getNumberOfSlaves(), "TASK_RUNNING");
+            new TasksResponse(esTasks, 3, "TASK_RUNNING");
```
Comment (sadovnikov) [inline system-test/src/main/java/org/apache/mesos/elasticsearch/systemtest/Main.java]: 3 will be difficult to understand. Should be replaced by cluster.getSlaves().length

File: system-test/src/systemTest/java/org/apache/mesos/elasticsearch/systemtest/FrameworkRoleSystemTest.java [hunk 3]
```diff
@@ -42,8 +42,8 @@ private void testMiniMesosReportsFrameworkRole(String role) throws UnirestExcept
         LOGGER.info("Started Elasticsearch scheduler on " + scheduler.getIpAddress() + ":" + getTestConfig().getSchedulerGuiPort());
 
         ESTasks esTasks = new ESTasks(TEST_CONFIG, scheduler.getIpAddress());
-        new TasksResponse(esTasks, CLUSTER.getConfig().getNumberOfSlaves());
-        new ElasticsearchNodesResponse(esTasks, CLUSTER.getConfig().getNumberOfSlaves());
+        new TasksResponse(esTasks, 3);
+        new ElasticsearchNodesResponse(esTasks, 3);
```
Comment (sadovnikov) [inline system-test/src/systemTest/java/org/apache/mesos/elasticsearch/systemtest/FrameworkRoleSystemTest.java]: 3 will be difficult to understand. Should be replaced by cluster.getSlaves().length

File: system-test/src/systemTest/java/org/apache/mesos/elasticsearch/systemtest/ZookeeperFrameworkSystemTest.java [hunk 4]
```diff
@@ -71,9 +71,9 @@ public void testZookeeperFramework_differentPath() throws UnirestException {
         CONTAINER_LIFECYCLE_MANAGEMENT.addAndStart(scheduler);
         ESTasks esTasks = new ESTasks(TEST_CONFIG, scheduler.getIpAddress());
 
-        new TasksResponse(esTasks, CLUSTER.getConfig().getNumberOfSlaves());
+        new TasksResponse(esTasks, 3);
```
Comment (sadovnikov) [inline system-test/src/systemTest/java/org/apache/mesos/elasticsearch/systemtest/ZookeeperFrameworkSystemTest.java]: 3 will be difficult to understand. Should be replaced by cluster.getSlaves().length

**Comment by philwinder** [comment]
AFAIK, the upgrade to MM 0.5.0 is the thing that should fix the system tests, as the old versions of MM didn't work with pre-docker 1.8. However, this doesn't pass on the PR build: https://ci.infra.container-solutions.com/view/All/job/ElasticsearchPullRequests/220/console
and doesn't even build on my Mac. I will start debugging this. If I get it working, I will address Viktor's comments.

File: system-test/src/main/java/org/apache/mesos/elasticsearch/systemtest/Main.java (hunk 2 above)
Comment (philwinder) [inline system-test/src/main/java/org/apache/mesos/elasticsearch/systemtest/Main.java]: This method does not exist in the new api, hence why he put three. Could consider adding. For now, setting number to static field.

**Comment by philwinder** [comment]
With the above commit, all of Viktor's issues have been addressed and code now compiles and system tests work on OSX. However the system tests still fail on Jenkins: https://ci.infra.container-solutions.com/job/ElasticsearchPullRequests/221/console
All tests are failing, so it must be something to do with the jenkins setup. Since that will not be fixed until after the holidays, I will merge this and assume it is working so work can continue.

**Pull request merged**